### ISSUE TITLE
get_entity_by_id() performance is poor in 0.11.0

### ISSUE THREAD

**Issue by yhmo** [milvus-io/milvus#4488]
**Please state your issue using the following template and, most importantly, in English.**

**Describe the bug**
get_entity_by_id() performance is poor in 0.11.0

**Steps/Code to reproduce behavior**
1. create a collection with a vector field of 2048 dimensions
2. insert 500000 vectors into the collection
3. call get_entity_by_id() to get one vector
result:
It cost 2~4 seconds

**Expected behavior**
In 0.10.x, with same dataset, it only cost 5~10 milliseconds to get one vector from collection. 

**Method of installation**
- [ ] Docker/cpu
- [ ] Docker/gpu
- [ ] Build from source

**Environment details**
- Hardware/Softwars conditions (OS, CPU, GPU, Memory)

- Milvus version (master or released version)

**Configuration file**
Settings you made in `server_config.yaml` or `milvus.yaml`

```yaml
paste-file-content-here
```

**Screenshots**

**Additional context**

**Comment by yhmo** [comment]
Script to reproduce(0.11.0):

```
import sys, getopt
import time
import random
import json
import threading

import numpy as np
from milvus import Milvus, Status, DataType


SERVER_ADDR = "127.0.0.1"
SERVER_PORT = '19530'

COLLECTION_NAME = "TEST6"
SEGMENT_ROW_COUNT = 500000

VECTOR_FIELD_NAME = "vec_float"
VECTOR_DIMENSION = 2048

COLLECTION_FIELDS = [
    {"name": VECTOR_FIELD_NAME, "type": DataType.FLOAT_VECTOR, "params": {"dim": VECTOR_DIMENSION}},
    ]

AUTO_ID = False

METRIC_TYPE = "L2"

MILVUS = Milvus(host=SERVER_ADDR, port=SERVER_PORT)


def create_collection():
    try:
        create_param = {
            "fields": COLLECTION_FIELDS,
            "segment_row_limit": SEGMENT_ROW_COUNT,
            "auto_id": AUTO_ID}

        status = MILVUS.create_collection(collection_name=COLLECTION_NAME, fields=create_param)
        print(COLLECTION_NAME, "created", status)
    except:
        print('creatpipe_collection exception')

def count_collection():
    try:
        row_count = MILVUS.count_entities(collection_name=COLLECTION_NAME)
        print(COLLECTION_NAME, "row count =", row_count)
    except:
        print('count_entities exception')


def gen_vec_list(nb, dim, seed=np.random.RandomState(1234)):
    xb = seed.rand(nb, dim).astype("float32")
    vec_list = xb.tolist()

    return vec_list

def generate_entities(count):
    entities = []
    for field in COLLECTION_FIELDS:
        name = field['name']
        type = field['type']

        if type == DataType.FLOAT_VECTOR:
            params = field['params']
            dim = params['dim']

            dd = gen_vec_list(nb=count, dim=dim)
            entities.append({"name": name, "type": type, "values": dd})

    return entities

def get_entity_by_ids(ids):
    fields = [VECTOR_FIELD_NAME]
    res = MILVUS.get_entity_by_id(collection_name=COLLECTION_NAME, ids=ids, fields=fields)
    return res


create_collection()

# insert entities
batch = 50
batch_count = 10000
ids = []
for i in range(batch):
    print("No.", i)
    entities = generate_entities(batch_count)
    entity_ids = [k + i * batch_count for k in range(batch_count)]
    ids = MILVUS.insert(collection_name=COLLECTION_NAME, entities=entities, ids=entity_ids)

MILVUS.flush(collection_name_array=[COLLECTION_NAME])

# get entity by id
target_ids = [1]
time_start = time.time()
res = get_entity_by_ids(ids=target_ids)
time_end = time.time()
print("get_entity_by_ids time cost:", time_end - time_start, 'sec')
```

**Comment by yhmo** [comment]
Script to verify the returned entities:

```
import time
import random

import numpy as np
from milvus import Milvus, Status, DataType


SERVER_ADDR = "127.0.0.1"
SERVER_PORT = '19530'

COLLECTION_NAME = "TEST"
SEGMENT_ROW_COUNT = 500000

VECTOR_FIELD_NAME = "vec_float"
VECTOR_DIMENSION = 2048

COLLECTION_FIELDS = [
    {"name": VECTOR_FIELD_NAME, "type": DataType.FLOAT_VECTOR, "params": {"dim": VECTOR_DIMENSION}},
    ]

AUTO_ID = False

METRIC_TYPE = "L2"

MILVUS = Milvus(host=SERVER_ADDR, port=SERVER_PORT)


def create_collection():
    try:
        create_param = {
            "fields": COLLECTION_FIELDS,
            "segment_row_limit": SEGMENT_ROW_COUNT,
            "auto_id": AUTO_ID}

        status = MILVUS.create_collection(collection_name=COLLECTION_NAME, fields=create_param)
        print(COLLECTION_NAME, "created", status)
    except:
        print('creatpipe_collection exception')

def count_collection():
    try:
        row_count = MILVUS.count_entities(collection_name=COLLECTION_NAME)
        print(COLLECTION_NAME, "row count =", row_count)
    except:
        print('count_entities exception')

def drop_collection():
    try:
        print("drop collection:", COLLECTION_NAME)
        status = MILVUS.drop_collection(collection_name=COLLECTION_NAME)
    except:
        print('drop_collection exception')

def generate_entities(count, seed=np.random.RandomState(1)):
    entities = []
    for field in COLLECTION_FIELDS:
        name = field['name']
        type = field['type']

        if type == DataType.FLOAT_VECTOR:
            params = field['params']
            dim = params['dim']

            xb = seed.rand(count, dim).astype("float32")
            dd = xb.tolist()
            entities.append({"name": name, "type": type, "values": dd})

    return entities

def get_entity_by_ids(ids):
    fields = [VECTOR_FIELD_NAME]
    res = MILVUS.get_entity_by_id(collection_name=COLLECTION_NAME, ids=ids, fields=fields)
    return res

drop_collection()
create_collection()

# insert entities
batch = 120
batch_count = 10000
entity_ids = []
entities = []
for i in range(batch):
    print("No.", i)
    entities = generate_entities(batch_count, seed=np.random.RandomState(i + 100))
    entity_ids = [k + i * batch_count for k in range(batch_count)]
    ids = MILVUS.insert(collection_name=COLLECTION_NAME, entities=entities, ids=entity_ids)

MILVUS.flush(collection_name_array=[COLLECTION_NAME])

# get entity by id
for k in range(1000):
    idx = random.randint(0, batch_count - 1)
    target_ids = entity_ids[idx:idx+1]
    time_start = time.time()
    res = get_entity_by_ids(ids=target_ids)
    time_end = time.time()
    res = res.dict()

    vector = res[0]['values'][0]
    target_entity = entities[0]['values'][idx]
    # print(vector)
    # print(target_entity)
    for i in range(VECTOR_DIMENSION):
        if vector[i] != target_entity[i]:
            print('Not equal!', vector[i], target_entity[i])

    print("get_entity_by_ids time cost:", time_end - time_start, 'sec')
```

**Comment by yhmo** [comment]
pr4489 will fix this issue in 0.11.1. After this fix, The time cost for each get_entity_by_id() in a 4GB file reduce from 2 seconds to 4~5 milliseconds.
